kosat-core/src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -389,13 +383,16 @@ class CDCL {
         }
     }
 
+    /**
+     * Reset the solver to the state before the last [solve].
+     */
     fun reset() {
-        level = 0
-        clearTrail(0)
-        qhead = trail.size
+        cancelUntil(0)
     }
 
-    // return current assignment of variables
+    /**
+     * Return the current assignment of variables.
+     */
     private fun getModel(): Model = Model(
```

Comment:
Consider caching the last SAT model. In this case, the comment should be modified. Note that in this case even if the incremental problem becomes UNSAT, the "last SAT model" is going to be still valid.
Another possibility is to return `Model?` (also consider caching here).